### PR TITLE
feat: add `pulumi cli` to autocomplete 

### LINKED ISSUE #669: [pulumi] CLI
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [ ] I would like to work on this.

### What CLI tool does this relate to?

pulumi/pulumi

### Which statement makes the most sense?

This completion spec does not exist

### Issue Details

https://www.pulumi.com/docs/reference/cli/

### PR DESCRIPTION
**Pull request by MishaKav**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature add [pulumi-cli](https://www.pulumi.com/docs/reference/cli/)

**What is the current behavior? (You can also link to an open issue here)**
No autocomplete for `pulumi-cli`

**What is the new behavior (if this is a feature change)?**
Add autocomplete to `pulumi-cli`

**Additional info:**
Resolves #669 
Spec has been added manually from the [official documentation](https://www.pulumi.com/docs/reference/cli/)

### CODE REVIEW

File: src/pulumi.ts [hunk 1]
```diff
@@ -0,0 +1,1604 @@
...
+    },
+  ],
+  options: [
+    {
+      name: ["--help", "-h"],
+      description: "Help for pulumi",
+    },
+  ],
+  // Only uncomment if pulumi takes an argument
+  // args: {}
```
Comment (QuiiBz) [inline src/pulumi.ts:1601]: Can you remove these comments?

File: src/pulumi.ts [hunk 2]
```diff
@@ -0,0 +1,1604 @@
...
+  name: ["-j", "--json"],
+  description: "Emit output as JSON",
+};
+
+const configFileOption: Fig.Option = {
+  name: "--config-file",
+  description:
+    "Use the configuration values in the specified file rather than detecting the file name",
+  args: {
+    name: "file",
```
Comment (QuiiBz) [inline src/pulumi.ts:26]: Looks like this argument can have a `template` of `filepaths`

File: src/pulumi.ts [hunk 3]
```diff
@@ -0,0 +1,1604 @@
...
+      default: "auto",
+      suggestions: ["always", "never", "raw", "auto"],
+    },
+  },
+  {
+    name: ["-C", "--cwd"],
+    description: "Run pulumi as if it had been started in another directory",
+    args: {
+      name: "path",
+      default: "./",
```
Comment (QuiiBz) [inline src/pulumi.ts:118]: Same as above but with a template of `folders`

File: src/pulumi.ts [hunk 4]
```diff
@@ -0,0 +1,1604 @@
...
+    name: "--profiling",
+    description:
+      "Emit CPU and memory profiles and an execution trace to '[filename].[pid].{cpu,mem,trace}', respectively",
+  },
+  {
+    name: "--tracing",
+    description:
+      "Emit tracing to the specified endpoint. Use the file: scheme to write tracing data to a local file",
+    args: {
+      name: "file",
```
Comment (QuiiBz) [inline src/pulumi.ts:151]: Same as above, I think this issue is repeating a few times

File: src/pulumi.ts [hunk 5]
```diff
@@ -0,0 +1,1604 @@
...
+    description:
+      "Allows destroying of dependent targets discovered but not specified in --target list",
+  },
+];
+
+const icon = "https://www.pulumi.com/logos/brand/avatar-on-white.svg";
+
+const completionSpec: Fig.Spec = {
+  name: "pulumi",
+  description: "",
```
Comment (QuiiBz) [inline src/pulumi.ts:220]: Can you add a description?

File: src/pulumi.ts [hunk 6]
```diff
@@ -0,0 +1,1604 @@
...
+              },
+            },
+            {
+              name: ["-h", "--help"],
+              description: "Help for cp",
+            },
+            {
+              name: "--path",
+              description:
+                "The key contains a path to a property in a map or list to set",
```
Comment (QuiiBz) [inline src/pulumi.ts:288]: This argument can also have a `template`

File: src/pulumi.ts [hunk 7]
```diff
@@ -0,0 +1,1604 @@
...
+          name: "--suppress-outputs",
+          description:
+            "Suppress display of stack outputs (in case they contain sensitive values)",
+        },
+        {
+          name: ["-h", "--help"],
+          description: "Help for import",
+        },
+      ],
+      args: { name: "[type] [name] [id]" },
```
Comment (QuiiBz) [inline src/pulumi.ts:514]: This should be split in multiple arguments:
```suggestion
      args: [
        {
          name: "type",
        },
        {
          name: "name",
        },
        ...
      ],
```

File: src/pulumi.ts [hunk 8]
```diff
@@ -0,0 +1,1604 @@
...
+        },
+        {
+          name: "--dir",
+          description:
+            "The location to place the generated project; if not specified, the current directory is used",
+          args: { name: "location" },
+        },
+        {
+          name: ["-f", "--force"],
+          description:
```
Comment (QuiiBz) [inline src/pulumi.ts:628]: Should we add `isDangerous`?

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks a lot for this PR!

File: src/pulumi.ts (hunk 2 above)
Comment (MishaKav) [reply src/pulumi.ts:27]: add whenever it's possible

File: src/pulumi.ts (hunk 3 above)
Comment (MishaKav) [reply src/pulumi.ts]: add whenever it possible

File: src/pulumi.ts (hunk 6 above)
Comment (MishaKav) [reply src/pulumi.ts]: it's a path to property inside the dictionary, not `filepath`.

File: src/pulumi.ts (hunk 4 above)
Comment (MishaKav) [reply src/pulumi.ts:164]: add whenever it possible

File: src/pulumi.ts (hunk 5 above)
Comment (MishaKav) [reply src/pulumi.ts]: done

File: src/pulumi.ts (hunk 7 above)
Comment (MishaKav) [reply src/pulumi.ts]: done

File: src/pulumi.ts (hunk 8 above)
Comment (MishaKav) [reply src/pulumi.ts:626]: added whenever it's possible

**Review by QuiiBz — APPROVED** [review]
LGTM, thank you so much for this huge contribution!

**Pull request merged**